dev/archery/archery/crossbow/reports.py

```diff
@@ -160,6 +160,29 @@ def url(self, query):
         repo_url = self.job.queue.remote_url.strip('.git')
         return '{}/branches/all?query={}'.format(repo_url, query)
 
+    def todayStr(self):
+        date = datetime.now()
+        return "{}-{}-{}".format(date.year, date.month, date.day)
+
+    def tasksToDict(self, date, tasks):
+        jsonTasks = []
+        for task_name, task in tasks.items():
+            jsonTasks.append({
+                "build" : task_name,
+                "link" : self.url(task.branch),
+                "status" : task.status().combined_state.upper(),
+                "timestamp" : date})
+
+        return jsonTasks
+
+    def getJsonTasks(self):
+        tasks = self.tasksToDict(self, self.todayStr(), self.job.tasks.items())
```

Comment:
FYI, I think this might have a bug. In general you don't need to pass `self` as an argument because `self.X` already implies that the first argument will be `self`

Reply:
sure ! I'll edit, my python knowledge is as good as my latin knowledge